### PR TITLE
[DOCS] Reformats URI search request

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45621.

This PR updates the URI search request API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Resources:
* [Search URI request API doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/search-uri-request.html)

Because of 7ae5fc8, this PR depends on #46057.

### CODE REVIEW

File: docs/reference/search/uri-request.asciidoc [hunk 1]
```diff
@@ -1,10 +1,121 @@
...
+GET twitter/_search?q=user:kimchy
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:twitter]
+
+
+[[search-uri-request-api-request]]
+==== {api-request-title}
+
+`GET {index}/_search?q={parameter}`
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]: In the Docs huddle, we agree to use the `<var>` convention going forward.

```suggestion
`GET /<index>/_search?q=<parameter>`
```

File: docs/reference/search/uri-request.asciidoc [hunk 2]
```diff
@@ -1,10 +1,121 @@
...
+[[search-uri-request-api-desc]]
+==== {api-description-title}
+
+A search request can be executed purely using a URI by providing request 
 parameters. Not all search options are exposed when executing a search
-using this mode, but it can be handy for quick "curl tests". Here is an
-example:
+using this mode, but it can be handy for quick "curl tests".
+
+
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]: Looks like we're missing the section for path parameters.

File: docs/reference/search/uri-request.asciidoc [hunk 3]
```diff
@@ -1,10 +1,121 @@
...
+[[search-uri-request-api-query-params]]
+==== {api-query-parms-title}
+
+`allow_partial_search_results`:: 
+  Set to `false` to return an overall failure if the request would produce
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]:
```suggestion
(Optional, boolean)  Set to `false` to return an overall failure if the request would produce 
```

File: docs/reference/search/uri-request.asciidoc [hunk 4]
```diff
@@ -1,10 +1,121 @@
...
+==== {api-query-parms-title}
+
+`allow_partial_search_results`:: 
+  Set to `false` to return an overall failure if the request would produce 
+  partial results. Defaults to true, which will allow partial results in the 
+  case of timeouts or partial failures. This default can be controlled using the 
+  cluster-level setting `search.default_allow_partial_results`.
+
+`analyze_wildcard`:: 
+  Should wildcard and prefix queries be analyzed or not. Defaults to `false`.
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]:
```suggestion
(Optional, boolean)  Should wildcard and prefix queries be analyzed or not. Defaults to `false`.
```

File: docs/reference/search/uri-request.asciidoc [hunk 5]
```diff
@@ -1,10 +1,121 @@
...
+  
+`analyzer`:: 
+  The analyzer name to be used when analyzing the query string.
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]:
```suggestion
(Optional, string) The analyzer name to be used when analyzing the query string.
```

File: docs/reference/search/uri-request.asciidoc [hunk 6]
```diff
@@ -1,10 +1,121 @@
...
+
+`batched_reduce_size`:: 
+  The number of shard results that should be reduced at once on the coordinating
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]:
```suggestion
  (Optional, integer) The number of shard results that should be reduced at once on the coordinating 
```

File: docs/reference/search/uri-request.asciidoc [hunk 7]
```diff
@@ -1,10 +1,121 @@
...
+  The analyzer name to be used when analyzing the query string.
+
+`batched_reduce_size`:: 
+  The number of shard results that should be reduced at once on the coordinating 
+  node. This value should be used as a protection mechanism to reduce the memory 
+  overhead per search request if the potential number of shards in the request 
+  can be large.
+
+`default_operator`:: 
+  The default operator to be used, can be `AND` or `OR`. Defaults to `OR`.
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]:
```suggestion
  (Optional, string) The default operator to be used, can be `AND` or `OR`. Defaults to `OR`.
```

File: docs/reference/search/uri-request.asciidoc [hunk 8]
```diff
@@ -1,10 +1,121 @@
...
+
+`df`:: 
+  The default field to use when no field prefix is defined within the query.
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]:
```suggestion
  (Optional, string) The default field to use when no field prefix is defined within the query.
```

File: docs/reference/search/uri-request.asciidoc [hunk 9]
```diff
@@ -1,10 +1,121 @@
...
+
+`explain`:: 
+  For each hit, contain an explanation of how scoring of the hits was computed.
```
Comment (jrodewig) [inline docs/reference/search/uri-request.asciidoc]: We'll need the `(Optional/Required, datatype)` for all the following parameters.

**Review by jrodewig — CHANGES_REQUESTED** [review]
I actually think this is the same API endpoint as https://www.elastic.co/guide/en/elasticsearch/reference/master/search-search.html.

We may just need to add any missing parameters and include an example in that documentation.

**Comment by szabosteve** [comment]
@jrodewig Yes, it is the same endpoint. Furthermore, [request body search](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/search-request-body.html) also uses the same endpoint. I thought about documenting them together, but it would end in an extremely long reference page. Should I still merge these three pages together and delete the standalone versions of this API?

**Review by debadair — COMMENTED** [review]
Left some editorial comments because I was checking out the query parameters. Do we want to add the search related ones to common parameters?

File: docs/reference/search/uri-request.asciidoc [hunk 10]
```diff
@@ -1,10 +1,121 @@
...
+
+[[search-uri-request-api-query-params]]
+==== {api-query-parms-title}
+
+`allow_partial_search_results`:: 
+  Set to `false` to return an overall failure if the request would produce 
+  partial results. Defaults to true, which will allow partial results in the 
+  case of timeouts or partial failures. This default can be controlled using the 
+  cluster-level setting `search.default_allow_partial_results`.
+
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/reference/search/uri-request.asciidoc]: @debadair @jrodewig I think it's worth to add the search-related query parameters to the common parameters. I'll do it in a separate PR and add both of you as reviewers.

**Comment by szabosteve** [comment]
Because of 7ae5fc8, this PR depends on #46057.

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks for putting this together @szabosteve=.

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Comment by szabosteve** [comment]
Backported to 7.x and 7.3.

**Comment by szabosteve** [comment]
Backported to 7.x and 7.3.

**Pull request merged**